.github/workflows/tests.yml

```diff
@@ -41,6 +41,11 @@ jobs:
         git diff
         # already well formated if 'git diff' doesn't output
         ! ( git diff |  grep -q ^ ) || return 1
+    - name: Test with gtest
+      run: |
+        git submodule init && git submodule update
+        cmake .
+        make && CTEST_OUTPUT_ON_FAILURE=1 make test
```

Comment:
Can we have this tests ran after the build of the package? Ideally, the build of the package can be conditioned to build tests and thus this step could be just to run the tests instead of bulding and testing